lib/chef/provider/chef_organization.rb

```diff
@@ -52,7 +52,9 @@ def whyrun_supported?
     new_resource.members.each do |user|
       if !existing_members.include?(user)
         converge_by "Add #{user} to organization #{new_resource.name}" do
-          rest.post("#{rest.root_url}/organizations/#{new_resource.name}/users/#{user}", {})
```

Comment:
Yep, for chef server 12 and future, it's: 

`rest.post("#{rest.root_url}/organizations/#{new_resource.name}/users", "username" => "username_to_associate" )`

Note that the requestor must be superuser for this to work.